api/src/main/java/com/epam/pipeline/controller/cluster/ClusterController.java

```diff
@@ -236,4 +238,17 @@ public void downloadNodeUsageStatisticsReport(
     public Result<List<NodeDisk>> loadNodeDisks(@PathVariable(value = NAME) final String name) {
         return Result.success(clusterApiService.loadNodeDisks(name));
     }
+
+    @PostMapping("/cluster/dnsrecord")
```

Comment:
What is the goal of using one method for `getOrCreate` and `delete` operations? If there is no some issues with this, I'd prefer to split these methods on all layers.